### PR TITLE
chore(ci): matrix build/test multiple Node.js versions

### PR DESCRIPTION
**Pull request by BethGriggs**
Seeing how CI fares here. 

I think we many want to consider testing on both Node.js 18 and 20 in the workflow, though.

### CODE REVIEW

**Review by nickboldt — APPROVED** [review]
LGTM but the real test will be downstream in showcase and then again in RHDH builds.

File: .github/workflows/pr.yaml [hunk 1]
```diff
@@ -30,3 +44,17 @@ jobs:
         uses: ./.github/actions/build
         with:
           args: '[HEAD^1]'
+
+  build20:
+    name: Build all packages
```
Comment (nickboldt) [inline .github/workflows/pr.yaml:49]:
```suggestion
    name: Build all packages with Node 20
```

File: .github/actions/build20/action.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      run: |
+        yarn install
+        if [[ $(git diff --name-only . || true) ]]; then
+          echo "After 'yarn install', workspace is dirty! The following files have changed:"
+          echo
+          git diff --name-only . || true
+          exit 42
+        fi
+
+    - name: Build all packages
```
Comment (nickboldt) [inline .github/actions/build20/action.yaml:30]:
```suggestion
    - name: Build all packages with Node 20
```

File: .github/workflows/pr.yaml [hunk 3]
```diff
@@ -17,6 +17,20 @@ jobs:
...
+        with:
+          fetch-depth: 0
+
+      - name: Test
+        uses: ./.github/actions/test20
+        with:
+          args: '[HEAD^1]'
+
   build:
     name: Build all packages
```
Comment (nickboldt) [inline .github/workflows/pr.yaml:35]:
```suggestion
    name: Build all packages with Node 18
```

**Comment by gashcrumb** [comment]
Hmmm, Playwright tests are skipped for this one, this would be good to [try locally](https://github.com/janus-idp/backstage-plugins/blob/main/CONTRIBUTING.md#ui-tests) but maybe we should just start running these on all PRs, WDYT @invincibleJai

File: .github/actions/build/action.yaml [hunk 4]
```diff
@@ -4,13 +4,16 @@ inputs:
   args:
     description: Backstage CLI repo build arguments
     default: '@janus-idp/*'
+  node-version:
+    description: Node.js version
+    required: true
```
Comment (nickboldt) [inline .github/actions/build/action.yaml:9]: no default value?
```suggestion
    required: true
    default: 20
```

File: .github/workflows/pr.yaml [hunk 5]
```diff
@@ -4,8 +4,11 @@ on: pull_request
 
 jobs:
   test:
-    name: Test
+    name: Test Node.js ${{ matrix.node-version }}
```
Comment (nickboldt) [inline .github/workflows/pr.yaml:7]:
```suggestion
    name: Test with Node.js ${{ matrix.node-version }}
```

File: .github/actions/test/action.yaml [hunk 6]
```diff
@@ -4,13 +4,16 @@ inputs:
   args:
     description: Backstage CLI repo build arguments
     default: '@janus-idp/*'
+  node-version:
+    description: Node.js version
+    required: true
```
Comment (nickboldt) [inline .github/actions/test/action.yaml:9]:
```suggestion
    required: true
    default: 20
```

**Comment by nickboldt** [comment]
Followup PR to fix the new test names in prow config: https://github.com/openshift/release/pull/51371

**Pull request merged**